packages/client-sdk-nodejs/src/internal/leaderboard-client.ts

```diff
@@ -0,0 +1,192 @@
+import {InternalLeaderboardClient} from '@gomomento/sdk-core/dist/src/internal/clients';
+import {
+  CredentialProvider,
+  InvalidArgumentError,
+  LeaderboardDelete,
+  LeaderboardFetch,
+  LeaderboardGetRank,
+  LeaderboardLength,
+  LeaderboardRemoveElements,
+  LeaderboardUpsert,
+  MomentoLogger,
+  SortedSetOrder,
+} from '@gomomento/sdk-core';
+import {LeaderboardClientProps} from '../leaderboard-client-props';
+import {normalizeSdkError} from '@gomomento/sdk-core/dist/src/errors';
+import {
+  validateCacheName,
+  validateLeaderboardName,
+  validateSortedSetRanks,
+  validateSortedSetScores,
+  validateSortedSetOffset,
+  validateSortedSetCount,
+} from '@gomomento/sdk-core/dist/src/internal/utils';
+import {Configuration} from '../config/configuration';
+import {delay} from '@gomomento/common-integration-tests';
+
+export class LeaderboardClient implements InternalLeaderboardClient {
+  private readonly configuration: Configuration;
```

Comment:
nit - this is the Configuration for the CacheClient. it has stuff on it that we probably won't want to have on the leaderboard client for day 1, and their respective configs may change over time, so we probably need to create a separate LeaderBoardConfiguration class/interface.